src/Console/Installer.php

```diff
@@ -0,0 +1,382 @@
+<?php
+namespace Muffin\Skeleton\Console;
+
+use Composer\Factory;
+use Composer\IO\IOInterface;
+use Composer\Json\JsonFile;
+use Composer\Script\Event;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+
+class Installer
+{
+
+    /**
+     * Author's name.
+     *
+     * @var string
+     */
+    private static $author;
+
+    /**
+     * Whether to create the `config` folder or not. Accepted
+     * values are `Y` and `N`.
+     *
+     * @var string
+     */
+    private static $configuration;
+
+    /**
+     * Plugin's description. Used in the `composer.json` and
+     * `README.md` files.
+     *
+     * @var string
+     */
+    private static $description;
+
+    /**
+     * The GitHub relative URL.
+     *
+     * @var string
+     */
+    private static $github;
+
+    /**
+     * Whether to create the `config/Migrations` folder or not.
+     * Accepted values are `Y` and `N`.
+     *
+     * @var string
+     */
+    private static $migrations;
+
+    /**
+     * The plugin's name (without vendor).
+     *
+     * @var string
+     */
+    private static $name;
+
+    /**
+     * The plugin's FQDN.
+     *
+     * @var string
+     */
+    private static $namespace;
+
+    /**
+     * The plugin's package name (as used by packagist.org).
+     *
+     * @var string
+     */
+    private static $package;
+
+    /**
+     * The plugin's full name (including vendor prefix).
+     *
+     * @var string
+     */
+    private static $plugin;
+
+    /**
+     * The plugin's vendor name.
+     *
+     * @var string
+     */
+    private static $vendor;
+
+    /**
+     * Pre-install hook.
+     *
+     * @param \Composer\Script\Event $event Composer script event.
+     * @return void
+     */
+    public static function preInstall(Event $event)
+    {
+        $io = $event->getIO();
+
+        static::$author = static::ask($io, 'What is your name ?', null, static::_gitConfig('user.name'));
```

Comment:
It might not be evident to users that the answer will be used to populate "author"